pelican/settings.py

```diff
@@ -660,3 +661,16 @@ def configure_settings(settings):
             continue            # setting not specified, nothing to do
 
     return settings
+
+
+def coerce_overrides(overrides):
+    coerced = {}
+    setting_types = {k: type(v) for k, v in DEFAULT_CONFIG.items()}
+    types_to_cast = {int, str}
+    for k, v in overrides.items():
+        setting_type = setting_types[k]
```

Comment:
You'll get an unclear error here if an unknown setting is passed. Better to check and provide a more helpful error message. Either log a warning and ignore or raise error and stop. I'm partial to both approaches.

Reply:
I've now added a logging.warning()